### PR TITLE
Use setup-macos-aarch64-builder for aarch64 CI pipeline

### PR DESCRIPTION
**Pull request by viirya**
## Which issue does this PR close?

A follow up to #9191.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -322,7 +322,7 @@ jobs:
         with:
           submodules: true
       - name: Setup Rust toolchain
-        uses: ./.github/actions/setup-macos-builder
+        uses: ./.github/actions/setup-macos-aarch64-builder
```
Comment (comphead) [inline .github/workflows/rust.yml:325]: should we remove ./.github/actions/setup-macos-builder ?

Comment (comphead) [reply .github/workflows/rust.yml:325]: EDIT: my bad this change for aarch64 only

**Review by comphead — APPROVED** [review]
lgtm thanks @viirya

**Comment by viirya** [comment]
Thanks @comphead

**Pull request merged**